server/controllers/gyms/getTopGyms.ts

```diff
@@ -0,0 +1,60 @@
+/* eslint-disable no-unused-vars */
+import { NextFunction, Response, Request } from 'express';
+import Sequelize, { Op } from 'sequelize';
+import { Image, Review } from '../../database/models';
+import Gym from '../../database/models/gyms';
+
+export default async function getAllGyms(req: Request, res: Response, next: NextFunction) {
+  try {
+    let gyms: any = await Gym.findAll({
+      subQuery: false,
+      attributes: [
+        'id',
+        'gym_name',
+        'logo',
+        'city',
+        'description',
+        'features',
+        [
+          Sequelize.literal(`(
+        SELECT AVG(review.rate)
+          FROM reviews AS review
+          WHERE
+              review.gym_id = gyms.id
+            )`),
+          'avg_rate',
+        ],
+      ],
+      include: [
+        { model: Image, required: false, attributes: ['id', 'pathUrl'], limit: 1 },
+        {
+          model: Review,
+          required: false,
+          attributes: [],
+        },
+      ],
+      limit: 3,
+      group: ['gyms.id'],
+      order: [Sequelize.literal('avg_rate')],
```

Comment:
here you 're not ordering by descending rate
```suggestion
      order: [[Sequelize.literal('avg_rate'), 'DESC']],
```